src/frontend/src/optimizer/plan_node/logical_topn.rs

```diff
@@ -419,6 +419,11 @@ impl ToStream for LogicalTopN {
                 "OFFSET without LIMIT in streaming mode".to_string(),
             )));
         }
+        if self.limit() == 0 {
+            return Err(RwError::from(ErrorCode::InvalidInputSyntax(
+                "LIMIT 0 in streaming mode".to_string(),
```

Comment:
Maybe we can also ban LIMIT 0 in batch mode, to keep batch & streaming consistent